Tasks/DeployAzureResourceGroup/DeployAzureResourceGroup.ps1

```diff
@@ -24,6 +40,21 @@ import-module Microsoft.TeamFoundation.DistributedTask.Task.DevTestLabs
 import-module Microsoft.TeamFoundation.DistributedTask.Task.Common
 
 $ErrorActionPreference = "Stop"
+
+#Find the deployment definition File
+Write-Host "Find-Files -SearchPattern $csmFile"
+$deploymentDefinitionFiles = Find-Files -SearchPattern "$csmFile"
+Write-Host "deploymentDefinitionFiles= $deploymentDefinitionFiles" -ForegroundColor Yellow
+#Ensure that at most a single deployment definition (.json) file is found
+$csmFile = Get-SingleFile $deploymentDefinitionFiles $csmFile
+
+#Find the deployment definition Parameter File
+Write-Host "Find-Files -SearchPattern $csmParametersFile"
+$deploymentDefinitionParametersFiles = Find-Files -SearchPattern "$csmParametersFile"
```

Comment:
csmParametersFile could a folder when left empty - how will Find-Files work on it?